mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt

```diff
@@ -120,16 +100,64 @@ abstract class AbstractBaseActivity : AppCompatActivity(), CoroutineScope {
-        var uiOptions = window.decorView.systemUiVisibility
-        val flags = (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-            or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
-            or View.SYSTEM_UI_FLAG_FULLSCREEN)
-        uiOptions = if (isEnabled && !forceNonFullscreen) {
-            uiOptions or flags
+        @Suppress("DEPRECATION")
+        fun checkFullscreenPreR() {
+            var uiOptions = window.decorView.systemUiVisibility
+            val flags = (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
+                or View.SYSTEM_UI_FLAG_FULLSCREEN)
+            uiOptions = if (isEnabled && !forceNonFullscreen) {
+                uiOptions or flags
+            } else {
+                uiOptions and flags.inv()
+            }
+            window.decorView.systemUiVisibility = uiOptions
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            val insetsController = window.insetsController ?: return
+            insetsController.systemBarsBehavior = BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+            if (isEnabled && !forceNonFullscreen) {
+                insetsController.hide(WindowInsets.Type.systemBars())
+            } else {
+                insetsController.show(WindowInsets.Type.systemBars())
+            }
+        } else {
+            checkFullscreenPreR()
+        }
+    }
+
+    private fun checkStatusBarColor() {
+        @Suppress("DEPRECATION") var uiOptions = window.decorView.systemUiVisibility
+        @Suppress("DEPRECATION") val flagsPreR = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
+        } else {
+            0
+        }
+        val flagsR: Int
+        val currentNightMode = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
+        val black = ContextCompat.getColor(this, R.color.black)
+        @ColorInt val windowColor = if (currentNightMode == Configuration.UI_MODE_NIGHT_YES) {
+            uiOptions = uiOptions and flagsPreR.inv()
+            flagsR = 0
+            resolveThemedColor(android.R.attr.windowBackground, black)
         } else {
-            uiOptions and flags.inv()
+            uiOptions = uiOptions or flagsPreR
+            flagsR = WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                resolveThemedColor(android.R.attr.windowBackground, black)
+            } else {
+                black
+            }
+        }
+        window.navigationBarColor = windowColor
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            window.insetsController
+                ?.setSystemBarsAppearance(flagsR, WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS)
+        } else {
+            @Suppress("DEPRECATION")
+            window.decorView.systemUiVisibility = uiOptions
```

Comment:
I find that method a bit hard to follow. I'd suggest doing handling of `windowColor` first, and then having two separate code blocks for flag handling pre-R and post-R. That way we don't need separate `flagsR` and `flagsPreR` variables (which don't make sense since only one of them will be used on a given device anyway).